alexandria/core/validations.py

```diff
@@ -101,3 +101,17 @@ def validate_file(self, data, context):
         data["mime_type"] = content_type_header
 
         return data
+
+    @validator_for(Document)
+    def validate_document(self, data, context):
+        if context["request"].method == "PATCH" and "category" in data:
+            category = data["category"]
+            document = context["view"].get_object()
+            if hasattr(document, "category") and (
```

Comment:
`hasattr` check should be unnecessary as a document should always have a category or you could just check `document.category is not None`

Reply:
I think i got some KeyErrors or something like this thats why i put this in but ill check again.

Reply:
I removed it